sql-plugin/src/main/scala/com/nvidia/spark/rapids/SparkShims.scala

```diff
@@ -163,4 +166,40 @@ trait SparkShims {
       exprId: ExprId,
       qualifier: Seq[String] = Seq.empty,
       explicitMetadata: Option[Metadata] = None): Alias
+
+  def replaceWithAlluxioPathIfNeeded(
+      conf: RapidsConf,
+      relation: HadoopFsRelation,
+      partitionFilters: Seq[Expression],
+      dataFilters: Seq[Expression]): FileIndex
+
+  /**
+   * Contains a set of paths that are considered as the base dirs of the input datasets.
+   * The partitioning discovery logic will make sure it will stop when it reaches any
+   * base path.
+   *
+   * By default, the paths of the dataset provided by users will be base paths.
+   * Below are three typical examples,
+   * Case 1) `spark.read.parquet("/path/something=true/")`: the base path will be
+   * `/path/something=true/`, and the returned DataFrame will not contain a column of `something`.
+   * Case 2) `spark.read.parquet("/path/something=true/a.parquet")`: the base path will be
+   * still `/path/something=true/`, and the returned DataFrame will also not contain a column of
+   * `something`.
+   * Case 3) `spark.read.parquet("/path/")`: the base path will be `/path/`, and the returned
+   * DataFrame will have the column of `something`.
+   *
+   * Users also can override the basePath by setting `basePath` in the options to pass the new base
+   * path to the data source.
+   * For example, `spark.read.option("basePath", "/path/").parquet("/path/something=true/")`,
+   * and the returned DataFrame will have the column of `something`.
+   */
+  def getBasePaths(
+    hadoopConf: Configuration,
+    basePathOption: Option[Path],
+    rootPaths: Seq[Path],
+    leafFiles: Seq[Path]): Set[Path]
+
+  def replacePartitionDirectoryFiles(
```

Comment:
I'm confused by this, I get there is a databricks one since ParittionDirectory but I don't see anything calling that version. the only place replacePartitionDirectoryFiles is called is from Spark300Shims. replaceWithAlluxioPathIfNeeded which is going to call into its local version of replacePartitionDirectoryFiles not the databricks version

Reply:
Spark301dbShims -> replaceWithAlluxioPathIfNeeded(in Spark300Shims) -> replacePartitionDirectoryFiles (overrided in Spark301dbShims)